PluginGenerator/PluginBuilder.cs

```diff
@@ -17,14 +17,34 @@ public class PluginBuilder
     {
         public const string SONARQUBE_API_VERSION = "4.5.2";
 
+        /// <summary>
+        /// The name of the package in which the plugin will be created
+        /// </summary>
+        private const string CorePluginPackageNameTemplate = "org.sonarqube.plugin.sdk.[PLUGIN_KEY]";
+
+        /// <summary>
+        /// The name of the class that SonarQube should load to discover the
+        /// extensions offered by the plugin
+        /// </summary>
+        private const string CorePluginEntryClassTemplate = "[PLUGIN_PACKGE].Plugin";
+
+        /// <summary>
+        /// Token that should be replaced with a list of the classes that exposed by
+        /// the plugin as a extensions.
+        /// Sample value: "org.sonarqube.plugin.sdk.MyRulesDefinition.class, myorg.otherExtension.class"
+        /// </summary>
+        private const string ExtensionListToken = "[CORE_EXTENSION_CLASS_LIST]";
```

Comment:
Previously, the RulesPluginGenerator did some of the work that should have been done by the core plugin builder class (e.g. adding the core jars, and performing token substitution). Some data items were also hard-coded e.g.the list of extensions that are exported. I've reduced the coupling between the core plugin builder and the rule plugin builder and added the ability to specify additional classes that will be exported.